agent/consul/autopilot_test.go

```diff
@@ -119,10 +113,19 @@ func testCleanupDeadServer(t *testing.T, raftVersion int) {
 }
 
 func TestAutopilot_CleanupDeadNonvoter(t *testing.T) {
-	dir1, s1 := testServer(t)
+	dir1, s1 := testServerWithConfig(t, func(c *Config) {
+		c.AutopilotConfig = &structs.AutopilotConfig{
+			CleanupDeadServers:      true,
+			ServerStabilizationTime: 100 * time.Millisecond,
+		}
+	})
 	defer os.RemoveAll(dir1)
 	defer s1.Shutdown()
 
+	// we have to wait for autopilot to be running long enough for the server stabilization time
+	// to kick in for this test to work.
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Why does this need to have the server stabilization time kick in? So that it won't immediately promote the server?